components/ui/NewsletterModal.tsx

```diff
@@ -0,0 +1,254 @@
+  const isOpen = !!cookies["DecoNewsletterModal"];
+
+  return { ...props, isOpen };
+};
+
+function InputNewsletter(
+  { name, placeholder, required, type }: InputNewletterProps,
+) {
+  return (
+    <input
+      name={name}
+      type={type}
+      class="input lg:h-12 h-9 px-5 join-item w-full mb-2.5 first:mt-5 border-2 border-neutral rounded-full placeholder:text-placeholder !outline-none lg:text-base text-xs"
+      placeholder={placeholder}
+      required={required}
+    />
+  );
+}
+
+function NewsletterModal(
+  {
+    isOpen,
+    form,
+    text,
+    modalSignExpiredDate,
+    modalCloseExpiredDate,
+  }: SectionProps<
+    ReturnType<typeof loader>
+  >,
+) {
+  const modalRef = useRef<HTMLDialogElement>(null);
+  const loading = useSignal(false);
+  const success = useSignal(false);
+
+  useEffect(() => {
+    if (isOpen !== true) {
+      modalRef.current?.showModal();
+    }
+  }, [isOpen]);
+
+  const handleSubmit: JSX.GenericEventHandler<HTMLFormElement> = async (e) => {
+    e.preventDefault();
+
+    try {
+      loading.value = true;
+
+      const email =
+        (e.currentTarget.elements.namedItem("email") as RadioNodeList)?.value;
+
+      let name = "";
+
+      if (form?.name?.show) {
+        name = (e.currentTarget.elements.namedItem("name") as RadioNodeList)
+          ?.value;
+      }
+
+      await subscribe({ email, name });
+    } finally {
+      loading.value = false;
+      success.value = true;
+
+      if (success.value === true) {
+        setCookieOnCloseModal("registed", modalSignExpiredDate);
+      }
+
+      setTimeout(() => {
+        success.value = false;
+      }, 5000);
+    }
+  };
+
+  const setCookieOnCloseModal = (
+    cookieValue: string,
+    expirationSeconds: number,
+  ) => {
+    // deno-lint-ignore no-var
+    var date = new Date();
+
+    date.setTime(date.getTime() + (expirationSeconds * 24 * 60 * 60 * 1000));
+    // deno-lint-ignore no-var
+    var expires = "expires=" + date.toUTCString();
+
+    document.cookie = "DecoNewsletterModal" + "=" + cookieValue + ";" +
+      expires +
+      ";path=/";
+  };
+
+  const emailInput = !form?.email?.show
+    ? (
+      <InputNewsletter
+        name="email"
+        required
+        type="email"
+        placeholder={form?.email?.placeholder || "E-mail"}
+      />
+    )
+    : null;
+
+  const nameInput = !form?.name?.show
+    ? (
+      <InputNewsletter
+        name="name"
+        type="text"
+        placeholder={form?.name?.placeholder || "Nome"}
+        required
+      />
+    )
+    : null;
+
+  return (
+    <>
+      <dialog id="my_modal_3" ref={modalRef} class="modal">
+        <form method="dialog" class="modal-box overflow-y-visible">
+          <div class="flex text-secondary-content justify-center items-center absolute right-2 -top-10">
+            <p class="font-normal">Não quero desconto</p>
+            <button
+              onClick={() => setCookieOnCloseModal("closed", 10)}
```

Comment:
O tempo ta fixo aqui no botao de fechar.
```suggestion
              onClick={() => setCookieOnCloseModal("closed", modalCloseExpiredDate)}
```